filewatcher/filewatcher.go

```diff
@@ -103,7 +108,7 @@ func (r *Result) watcherLoop(
 			case fsnotify.Create, fsnotify.Write:
 				// Wrap with an anonymous function to make sure that defer works.
 				func() {
-					f, err := os.Open(path)
+					f, err := limitopen.OpenWithLimit(path, softLimit, hardLimit)
 					if err != nil {
 						logger.Log(context.Background(), "filewatcher: I/O error: "+err.Error())
```

Comment:
The soft limit is logged to sentry, but the hard limit won't be?

Reply:
hard limit violation is a hard failure, which is already handled (for the first parser call it will be returned to the caller, for later calls it's handled by logger passed in, the logger passed into filewatcher is usually `log.ErrorWithSentryWrapper`, see https://github.com/reddit/baseplate.go/issues/365).